packages/hardhat/deploy/99_generateTsAbis.ts

```diff
@@ -27,20 +27,43 @@ function getContractNames(path: string) {
     .map(dirent => dirent.name.split(".")[0]);
 }
 
-function getInheritedFunctions(sources: Record<string, any>, contractName: string) {
-  const inheritedFunctions = {} as Record<string, any>;
+function getAcutalSourcesForContract(sources: Record<string, any>, contractName: string) {
   for (const sourcePath of Object.keys(sources)) {
     const sourceName = sourcePath.split("/").pop()?.split(".sol")[0];
     if (sourceName === contractName) {
-      continue;
+      const contractContent = sources[sourcePath].content as string;
+      const regex = /contract\s+(\w+)\s+is\s+([^{}]+)\{/;
```

Comment:
Also tested this regex nicely with different cases eg ;

`contract A is      B,C{`

`contract A is B,C{`

`contract A is B,C      {` 

`contract A     is  B,                       C {` 

etc just to be sure the spaces don't mess it up and it nicely returns  [B,C] every time 🙌